[Srinivasan Neharikha] iP

Neha-5678 opened the pull request:
# DukeTaskHandler

> "Your mind is for having ideas, not holding them.”
> - David Allen [source](https://dansilvestre.com/productivity-quotes/)

DukeTaskHandler makes it much easier to plan and manage your tasks on a daily basis. It's 

- easy to use
- easy to learn 
- very ~~fast~~ responsive

To start using the application, 

1. download the application
2. open it
3. key in your tasks
4. Enjoy as DukeTaskHandler manages everything for you :grinning:

Most importantly, DukeTaskHandler is **free** :grin:

*Features*:

- [ ] Add tasks
- [ ] Delete tasks
- [ ] Indicate tasks as done
- [ ] Find tasks

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
Here is an inline code:
`private boolean isHappy = true;`

Commit:
Add Gradle support

Commit:
Increment1

Commit:
Increment2

Commit:
Increment3

Commit:
Increment-4

Commit:
A-TextUiTesting

Commit:
Increment5

Commit:
Increment6

Commit:
Create separate files for nested classes

Commit:
Add Increment-7 and Increment-8

Commit:
Add more OOP

Commit:
Divide classes into packages

Commit:
Add JUnit Tests

Commit:
Add JavaDocs

Commit:
Modify code to follow coding standards

Commit:
Add find task feature

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'add-gradle-support' of https://github.com/Neha-5678/ip

Commit:
Set up Gradle in project

kvihashini commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,40 @@
+package duke;
+
+import duke.command.*;
```
Imported classes from the duke package could be explicitly stated as per [here](https://se-education.org/guides/conventions/java/intermediate.html#statements). I saw another instance in the other classes (eg. Storage.java) :-)

kvihashini commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,40 @@
+     * Returns the command to be executed in response to the user's command.
+     *
+     * @param command Input by the user.
+     * @return
```
Maybe it would be better for javadocs to include [description](https://se-education.org/guides/conventions/java/intermediate.html#comments) of what the method returns (eg. returns the command to be executed)

kvihashini commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    /**
+     * Gets all the tasks in the hard disk
```
Your code is really clean so i'm looking closely, but remember to include [punctuations](https://se-education.org/guides/conventions/java/intermediate.html#comments) after each description line!

kvihashini commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,64 @@
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import duke.Ui;
```
The [ordering](https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements) of the import statements in the all the classes can be consistent so that it's easier to browse the list :-)

kvihashini commented on src/main/java/duke/command/BlahCommand.java:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
I think you might have missed out on the header comments for the Command classes haha but remember to include them for all public methods :-)

kvihashini commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,34 @@
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+    public Integer getNumberOfTasks() {
```
could include javadoc header comments for public methods :-)

kvihashini reviewed (commented):
LGTM! good job! just a few nits to fix :-)

Neha-5678 committed:
Add GUI to Duke

Tetrerox commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,161 @@
+     * @throws FileNotFoundException If file is not found.
+     * @throws DukeException1 If there is an error in creating a todo, event, deadline task.
+     */
+    public void getContent(String filePath, ArrayList<Task> list) throws FileNotFoundException, DukeException1 {
```
Try to avoid naming classes/variables names with numbers (DukeException1) at the back, as stated [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/topics.html), under W4.6d.

Tetrerox commented on src/main/java/duke/exceptions/DukeException1.java:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Exception that is thrown when there are errors encountered in the Duke application.
+ */
+public class DukeException1 extends Exception {
```
Same comment here as above. Try to avoid naming classes/variables with numbers (DukeException1) at the back, as stated here, under W4.6d.

Tetrerox reviewed (commented):
Overall, good job on the code! 👍 I found it pretty difficult to find coding standards or naming violations and most of them were already mentioned by the previous reviewer.

Neha-5678 committed:
Add Assertions

There were assumptions made when writing the code.

Including Assertion statements allows us to test these assumptions.

Let's test the assumptions and stop the program if they do not hold.

Including assertion stops the program abruptly and prevents more errors.

Neha-5678 committed:
Merge pull request #1 from Neha-5678/branch-A-Assertions

Add Assertions

Neha-5678 committed:
Merge branch 'master' of https://github.com/Neha-5678/ip

Neha-5678 committed:
Merge branch 'master' into branch-A-CodeQuality

Neha-5678 committed:
Add CodeQuality

Code has deep nesting and does not follow SLAP.

This makes the code less readable and difficult to follow.

Modify code to avoid nesting and follow SLAP to improve readability.

Let's pull out relevant parts to rewrite as methods to follow SLAP.

Neha-5678 committed:
Merge pull request #2 from Neha-5678/branch-A-CodeQuality

Branch a code quality

Neha-5678 committed:
Merge branch 'master' of https://github.com/Neha-5678/ip

Neha-5678 committed:
Add D-Notes

The duke application does not have a feature to pen down notes.

Adding this feature will help users to remember some details.

Let's add a notes functionality to cater to users' needs.

Neha-5678 committed:
Merge branch 'branch-BCD-Extension'

Neha-5678 committed:
Add A-UserGuide

Neha-5678 committed:
Add delete functionality for notes

Neha-5678 committed:
Add more exceptions and error handling messages

Neha-5678 committed:
Add Javadoc comments

Neha-5678 committed:
Edit user guide

Neha-5678 committed:
Fix import statement to follow coding style

Neha-5678 committed:
Edit Javadoc comments

Neha-5678 committed:
Fix more coding style errors
